libs/damap/src/lib/interceptors/auth.interceptor.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpInterceptor,
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { AuthService } from '../auth/auth.service';
+import { FeedbackService } from '../services/feedback.service';
+import { OAuthService } from 'angular-oauth2-oidc';
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+  constructor(
+    private authService: AuthService,
+    private oAuthService: OAuthService,
+    private feedbackService: FeedbackService,
+  ) {}
+
+  intercept(
+    req: HttpRequest<any>,
+    next: HttpHandler,
+  ): Observable<HttpEvent<any>> {
+    const accessToken = this.oAuthService.getAccessToken();
+    if (accessToken == null) {
+      return next.handle(req);
+    }
+
+    const date = new Date(this.oAuthService.getAccessTokenExpiration());
+    const timeLeft = date.getTime() - new Date().getTime();
+
+    if (timeLeft < 5) {
+      alert('Your session has expired. Please log in again.');
+      this.authService.logout();
```

Comment:
suggestion: would it make sense to let the user try to login again? A user might have entered some data which would be lost by simply logging out.

Reply:
Due to the redirect when logging in, the data would be lost anyway, or am i missing something.

Reply:
I was thinking about the following:

There is a function called `initLoginFlowInPopup`in the `OAuthService`. We could allow the user to re-login within the popup:
- If successful, continue with the request as user is logged in again.
- If not successful, show popup again
- If user closes popup manually, logout and redirect

This assumes that a manual close can be recognized (to be investigated). We have to look at the specifics of the function to get a better picture of what is possible.